pkg/downloader/chart_downloader.go

```diff
@@ -210,12 +212,14 @@ func (c *ChartDownloader) ResolveChartVersion(ref, version string) (*url.URL, er
 	if err != nil {
 		return u, err
 	}
-	if r != nil && r.Config != nil && r.Config.Username != "" && r.Config.Password != "" {
-		c.Options = append(c.Options, getter.WithBasicAuth(r.Config.Username, r.Config.Password))
-	}
 
-	if r.Config.CertFile != "" || r.Config.KeyFile != "" || r.Config.CAFile != "" {
```

Comment:
Was missing null check from previous pr.

Reply:
I didn't find that null check meaningful, as `r` could only be null if `&{ ... }` fails (out of memory or some such), and `r.cfg` is the `rc` from the argument passed in to `repo.NewChartRepository()`, which can not be null at that point.

Reply:
Makes sense, sounds like this could be cleaned up by removing the check entirely. Just wanted to be consistent with guarding in the case it's necessary.

Reply:
Indeed, there were quite a few points that could be cleaned up throughout, but didn't want to introduce to many unrelated changes..